pkg/epp/scheduling/types/types.go

```diff
@@ -84,12 +101,18 @@ type PodMetrics struct {
 	*backendmetrics.Metrics
 }
 
-func NewSchedulingContext(ctx context.Context, req *LLMRequest, pods []Pod) *SchedulingContext {
-	logger := log.FromContext(ctx).WithValues("request", req)
+func NewSchedulingContext(ctx context.Context, req *LLMRequest, resp *LLMResponse, pods []Pod) *SchedulingContext {
+	var logger logr.Logger
+	if req != nil {
+		logger = log.FromContext(ctx).WithValues("request", req)
+	} else {
+		logger = log.FromContext(ctx).WithValues("response", resp)
+	}
```

Comment:
maybe it makes sense to separate SchedulingRequestContext and SchedulingResponseContext.
for example, when using context for response, do you use PodSnapshot?
would be good if each  gets only the data it uses and not super-object that contains all.